### PR TITLE
AST-1624 - fix - Blog page Meta Settings not work as it is the archive page

### PR DESCRIPTION
**Pull request by yssambare12**
### Description

- On Shop/blog pages the meta settings won't work.
- As it is the archive page, can we add any notice into the Astra Meta settings.
- As users might be confused about this why the Meta settings are not working for the Blog page?

- After Discussion decided to make Disable the footer & Disable the header working & Hide the remaining option from the archive pages.

### How has this been tested?

1. Go To  WordPress setting → Reading → 
2. Select a static page inside that selects any page as a recent post page.
3. Check on the particular page, from there meta setting options are hidden.
4. Content Layout & sidebar options are not visible on the selected page.
5. Also, check to the disabled header & footer are working on the particular selected archive page

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -23,6 +23,8 @@ v3.9.2 (Unreleased)
 - Fix: The Menu get disappear on the Android mobile device when you select Header Type is set as Dropdown.
 - Fix: Footer - Footer global background color and padding is not working in the frontend.
 - Fix: Spectra plugin settings page link redirect to old 'uag' admin page.
+- Fix: Disable header & footer not working on archive pages.
+
```
Comment (imnavanath) [inline changelog.txt:27]:
```suggestion
```

Comment (yssambare12) [reply changelog.txt:27]: Updated

File: inc/metabox/class-astra-meta-boxes.php [hunk 2]
```diff
@@ -964,6 +980,23 @@ public static function post_meta_options() {
 	}
 }
 
+/**
+ * Footer disable on archive pages.
+ *
+ * @param bool $display_footer for controling the header and footer enable/disable options.
+ *
+ * @since 3.9.1
```
Comment (imnavanath) [inline inc/metabox/class-astra-meta-boxes.php:988]:
```suggestion
 * @since x.x.x
```

File: inc/metabox/class-astra-meta-boxes.php [hunk 3]
```diff
@@ -475,6 +475,20 @@ public function load_scripts() {
...
+			global $post;
+			$astra_hide_content_layout_theme_settings = false;
+			$astra_hide_sidebar_theme_settings        = false;
+			if ( $post ) {
+				$page_for_posts = get_option( 'page_for_posts' );
+
+				if ( $post->ID == $page_for_posts ) {
+
+					$astra_hide_content_layout_theme_settings = true;
+					$astra_hide_sidebar_theme_settings        = true;
```
Comment (imnavanath) [inline inc/metabox/class-astra-meta-boxes.php:488]: No need of separate 2 variables, as we want to hide both options together we can directly hide them via single variable.

File: changelog.txt [hunk 4]
```diff
@@ -25,6 +25,7 @@ v3.9.2
 - Fix: The Menu gets disappeared on the Android mobile device when you select Header Type is set as Dropdown.
 - Fix: Footer - Footer global background color and padding is not working in the frontend.
 - Fix: Spectra plugin settings page link redirect to old 'uag' admin page.
+- Fix: Disable header & footer not working on archive pages.
```
Comment (imnavanath) [inline changelog.txt:28]: This line does not clear point, something for blog page should be there.

File: inc/metabox/class-astra-meta-boxes.php (hunk 3 above)
Comment (yssambare12) [reply inc/metabox/class-astra-meta-boxes.php:488]: Ok! I will update this

File: inc/metabox/class-astra-meta-boxes.php (hunk 2 above)
Comment (yssambare12) [reply inc/metabox/class-astra-meta-boxes.php]: > 

Updated.

File: inc/metabox/class-astra-meta-boxes.php (hunk 3 above)
Comment (yssambare12) [reply inc/metabox/class-astra-meta-boxes.php]: Updated.

File: changelog.txt (hunk 4 above)
Comment (yssambare12) [reply changelog.txt]: Updated.

File: inc/metabox/class-astra-meta-boxes.php [hunk 5]
```diff
@@ -475,6 +475,17 @@ public function load_scripts() {
 
 			wp_enqueue_script( 'astra-meta-settings' );
 			$astra_ext_extension_class_exists = class_exists( 'Astra_Ext_Extension' ) ? true : false;
+			
+			global $post;
+			
+			$ast_content_layout_sidebar = false;
+			if ( $post ) {
+				$page_for_posts = get_option( 'page_for_posts' );
+				if ( $post->ID == $page_for_posts ) {
```
Comment (imnavanath) [inline inc/metabox/class-astra-meta-boxes.php:484]:
```suggestion
				if ( $post->ID === $page_for_posts ) {
```

File: inc/metabox/extend-metabox/src/settings.js [hunk 6]
```diff
@@ -22,6 +22,8 @@ const MetaSettings = props => {
 
     const openModal = () => setOpen( true );
     const closeModal = () => setOpen( false );
+	
+	const is_hide_contnet_layout_sidebar = astMetaParams.is_hide_contnet_layout_sidebar;
```
Comment (neerajmasai) [inline inc/metabox/extend-metabox/src/settings.js:26]: `const is_hide_content_layout_sidebar = astMetaParams.is_hide_content_layout_sidebar;`

Comment (neerajmasai) [reply inc/metabox/extend-metabox/src/settings.js:26]: Similarly, variable naming can be fixed wherever used.

File: changelog.txt [hunk 7]
```diff
@@ -5,7 +5,7 @@ v3.9.3 (Unreleased)
 - Improvement: WooCommerce - Added product image on cart page for tablet users.
 - Fix: WooCommerce - Content background color preview not working in customizer preview.
 - Fix: WooCommerce - Shop - Sale badge text is misaligned on the frontend.
-- Fix: Block Editor - Astra Meta settings does not save when Custom Fields option is enabled from the preferences.
+- Fix: On blog page Astra meta setting not work as it is the archive page.
```
Comment (imnavanath) [inline changelog.txt:8]: @yssambare12 why changelog is replaced? Both are different points correct? You can add on a new line. 
Also please rephrase it, that should not be very geeky keep it simple & meaningful.

Comment (imnavanath) [inline changelog.txt:8]:
```suggestion
- Fix: Block Editor - Astra Meta settings does not save when Custom Fields option is enabled from the preferences.
- Fix: Disabled Astra meta setting for Blog page which does not not work from metabox, as it acts like a archive page on the frontend.
```

**Review by imnavanath — APPROVED** [review]

**Pull request merged**